advice-dont-ask-about-solutions.md

```diff
@@ -0,0 +1,56 @@
+# Don't ask about solutions[^1]
+
+You may have been sent this document because **you asked about solutions** to a problem,
+rather than the problem itself, or without explaining your original problem.
+This is called ["The XY Problem"](https://en.wikipedia.org/wiki/XY_problem)
+
+We often encounter problems that appear to have multiple, possible solutions. These
+solutions themselves however could cause more problems, which in turn have multiple
+solutions, which in turn may cause more problems...
+
+What may have happened here is that you just asked a question that mostly consists
+of what you tried or waht you're trying to do, rather than giving context on **why
+it is you think you have to do this**. For example,
+
+> You wanted to do <X>.
+> You ran into a problem with <X> or don't know how to do <X>.
+> You figured, you can solve <X> by doing <Y>.
+> You ran into a problem with <Y> or don't know how to do <Y>.
+> You asked for help with <Y>.
+> Others help you with <Y> but are confused, because they don't know about <X>.
+
+This leads to enormous amounts of wasted time and energy, both on your part, and
+on the part of those trying to help you.
+
+If you come to someone with a solution rather than the problem, you're denying them:
+- the chance to evaluate if your solution may solve the problem
+- the context on why this problem occurred and what you did
+- reproduction steps that they may recognize
+
+In the best case scenario, you may have mentioned your original problem, but spent
+more time describing the solution. **Instead, explain how you ran into the problem,
+and set your solution aside, so others have a chance to arrive at the same point
+you did.**
+
+In the worst case scenario, you may have completely omitted information on your
+original problem, and you may have been sent this document because the issue you
+posted above hints at another problem. **Instead, explain the context of what you
+did before you started working on what you posted about.**
+
+If you're not sure what to ask about, think back of when you started your task, and
+explain:
+> I started to work on <task> and wanted to achieve <goal>.
+> Instead of <expectation>, I encountered <problem>.
+
+If you talk about your original goals and problems rather than your later solutions,
+you're telling others the full story of what happened and you may save yourself and
+them some time having to hunt down where your problem originally started.
+
+### tl;dr
```

Comment:
I think the TL;DR's would be better at the start. So "You have been sent this because ..." -> TL;DR -> long explanation.